File: typescript/primes.ts
function isprime(n: number): boolean {
    if (n == 2 || n == 3) return true;
    if (n % 2 == 0 || n % 3 == 0) return false;
    for (let i = 5; i < Math.sqrt(n); i += 6) if (n % i == 0 || n % (i+2) == 0) return false;
    return true;
}

// driver code
for (let i = 1; i < 1000; i++)
    if (isprime(i))
        console.log("prime number found: " + i);

File: python/primes.py
import math

def isprime(n):
    if n == 2 or n == 3:
        return True
    if n % 2 == 0 or n % 3 == 0:
        return True
    for i in range(5, math.sqrt(n)+1, 6):
        if n % i == 0:
            return True
    return False

# driver code
for i in range(1, 1000):
    if isprime(i):
        print("prime number found: %d", i)